### PR TITLE
Add tests for cache adapter

### PR DESCRIPTION
**Pull request by socsieng**

### CODE REVIEW

File: packages/tests-unit/tests/adapters/cache.test.ts [hunk 1]
```diff
@@ -13,3 +15,471 @@ describe("hasCacheExtension", () => {
...
+        incrementalCache.get.mockResolvedValueOnce({
+          value: {
+            type: "route",
+            body: "{}",
+          },
+          lastModified: Date.now(),
+        });
+        globalThis.isNextAfter15 = true;
+
+        const result = await cache.get("key", { kindHint: "app" });
```
Comment (conico974) [inline packages/tests-unit/tests/adapters/cache.test.ts:186]: next 15 is not using `kindHint` anymore, it uses `kind`

File: packages/tests-unit/tests/adapters/cache.test.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import { hasCacheExtension } from "@opennextjs/aws/adapters/cache.js";
+/* eslint-disable sonarjs/no-duplicate-string */
+import S3Cache, { hasCacheExtension } from "@opennextjs/aws/adapters/cache.js";
```
Comment (conico974) [inline packages/tests-unit/tests/adapters/cache.test.ts:2]: The import is not correct anymore

**Review by conico974 — COMMENTED** [review]
I added a few comment

File: packages/tests-unit/tests/adapters/cache.test.ts (hunk 2 above)
Comment (socsieng) [reply packages/tests-unit/tests/adapters/cache.test.ts:2]: In what sense? I couldn't see any meaningful differences that would make the import incorrect: https://github.com/opennextjs/opennextjs-aws/compare/178ab2b1c95701a5f20aec107acf1fe1c6e3d9be...main#diff-c8326fdbbc9eb7b098d401bf8245a207a236441bae48edf1597f313afd7fa1c6

File: packages/tests-unit/tests/adapters/cache.test.ts [hunk 3]
```diff
@@ -13,3 +15,471 @@ describe("hasCacheExtension", () => {
...
+        const result = await cache.get("key", { kindHint: "fetch" });
+
+        expect(getFetchCacheSpy).toHaveBeenCalled();
+        expect(result).toBeNull();
+      });
+    });
+
+    describe("incremental cache", () => {
+      it.each(["app", "pages", undefined])(
+        "Should retrieve cache from incremental cache when hint is not fetch: %s",
```
[the remark this answers is not in the document]
Comment (socsieng) [reply packages/tests-unit/tests/adapters/cache.test.ts:132]: `%s` is used to differentiate between a few different test inputs defined above: `it.each(["app", "pages", undefined])`.

Example output:

```
✓ incremental cache (13)
  ✓ Should retrieve cache from incremental cache when hint is not fetch: app
  ✓ Should retrieve cache from incremental cache when hint is not fetch: pages
  ✓ Should retrieve cache from incremental cache when hint is not fetch: undefined
```

I think this is still required.

File: packages/tests-unit/tests/adapters/cache.test.ts [hunk 4]
```diff
@@ -13,3 +15,471 @@ describe("hasCacheExtension", () => {
...
+            rsc: "rsc",
+            meta: {
+              status: 200,
+            },
+          },
+          lastModified: Date.now(),
+        });
+        globalThis.isNextAfter15 = true;
+
+        const result = await cache.get("key", { kindHint: "app" });
```
[the remark this answers is not in the document]
Comment (socsieng) [reply packages/tests-unit/tests/adapters/cache.test.ts:268]: Based on the implementation of `isFetchCache`, this is not a valid test scenario for 15. Will remove this test.

File: packages/tests-unit/tests/adapters/cache.test.ts [hunk 5]
```diff
@@ -13,3 +15,471 @@ describe("hasCacheExtension", () => {
...
+            json: {},
+            meta: {
+              status: 200,
+            },
+          },
+          lastModified: Date.now(),
+        });
+        globalThis.isNextAfter15 = true;
+
+        const result = await cache.get("key", { kindHint: "pages" });
```
[the remark this answers is not in the document]
Comment (socsieng) [reply packages/tests-unit/tests/adapters/cache.test.ts:323]: Same as above. Will remove this test.

File: packages/tests-unit/tests/adapters/cache.test.ts (hunk 2 above)
Comment (conico974) [reply packages/tests-unit/tests/adapters/cache.test.ts:2]: My bad this is correct

File: packages/tests-unit/tests/adapters/cache.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (conico974) [reply packages/tests-unit/tests/adapters/cache.test.ts:164]: Ho yeah right, i missed the each

**Pull request merged**